### PR TITLE
Add support for _codehash in scilla contracts

### PR DESCRIPTION
**Pull request by joseph-zilliqa**
* Fixes https://github.com/Zilliqa/zq2/issues/1292

* Add support for _codehash in scilla contracts

* Enable test scilla/Codehash.ts

### CODE REVIEW

**Comment by JamesHinshelwood** [comment]
cc @saeed-zil @chetan-zilliqa 

Is there something to add at the execution side as well as the API side?

**Review by chetan-zilliqa — APPROVED** [review]
We are good.There is no api that returns the code hash.

File: zilliqa/src/scilla.rs [hunk 1]
```diff
@@ -715,7 +718,19 @@ impl ActiveCall {
                 Ok(Some((val, "ByStr20".to_owned())))
             }
             "_codehash" => {
-                todo!()
+                let code_bytes = match &account.account.code {
+                    Code::Evm(bytes) => bytes.clone(),
```
Comment (JamesHinshelwood) [inline zilliqa/src/scilla.rs:722]: Double check the behaviour in ZQ1, but I'd guess we should treat EVM accounts as if they had no Scilla code.

Comment (joseph-zilliqa) [reply zilliqa/src/scilla.rs:722]: It appears that ZQ1 has the same behaviour

**Review by JamesHinshelwood — APPROVED** [review]
cc @saeed-zil for potential conflicts with #1350

**Review by JamesHinshelwood — APPROVED** [review]
Thanks. Sorry for one last comment - Can we add an entry at the top of `CHANGELOG.md` documenting this change?

**Review by JamesHinshelwood — APPROVED** [review]

**Pull request merged**